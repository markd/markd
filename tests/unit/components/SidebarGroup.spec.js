import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '@/components/SidebarGroup.vue'

import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

const store = new Vuex.Store({
  state: {
    title: 'markd',
    pinned: [
      {
        icon: 'cloud_queue',
        name: 'All Bookmarks'
      },
      {
        icon: 'favorite_border',
        name: 'Favourite Bookmarks'
      }
    ],
    collections: [
      {
        colour: 'red',
        name: 'Web Development'
      },
      {
        colour: 'green',
        name: 'Spicy Memes'
      }
    ]
  }
})

describe('SidebarGroup.vue', () => {
  const propsData = {
    title: 'test'
  }

  it('renders', () => {
    const wrapper = shallowMount(Component, {
      localVue,
      store,
      propsData
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('display title', () => {
    const wrapper = shallowMount(Component, {
      localVue,
      store,
      propsData
    })

    expect(wrapper.find('.sidebarGroup__title').text()).toBe(propsData.title)
  })
})
