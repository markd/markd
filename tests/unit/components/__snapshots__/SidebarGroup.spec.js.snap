// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarGroup.vue renders 1`] = `
<div class="sidebarGroup">
  <h2 class="sidebarGroup__title">test</h2>
  <div class="sidebarGroup__content"></div>
</div>
`;
